src/main/java/nu/studer/teamcity/buildscan/internal/cleanup/BuildScanCleanupController.java

```diff
@@ -0,0 +1,40 @@
+package nu.studer.teamcity.buildscan.internal.cleanup;
+
+import jetbrains.buildServer.controllers.BaseAjaxActionController;
+import jetbrains.buildServer.serverSide.auth.Permission;
+import jetbrains.buildServer.serverSide.auth.SecurityContext;
+import jetbrains.buildServer.web.openapi.ControllerAction;
+import jetbrains.buildServer.web.openapi.WebControllerManager;
+import org.jdom.Element;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static jetbrains.buildServer.serverSide.auth.AuthUtil.hasGlobalPermission;
+
+public class BuildScanCleanupController extends BaseAjaxActionController implements ControllerAction {
+    private final CustomDataStorageBuildScanCleaner customDataStorageCleaner;
+    private final SecurityContext securityContext;
+
+    public BuildScanCleanupController(@NotNull WebControllerManager controllerManager,
+                                      @NotNull CustomDataStorageBuildScanCleaner customDataStorageCleaner,
+                                      @NotNull SecurityContext securityContext) {
+        super(controllerManager);
+        this.customDataStorageCleaner = customDataStorageCleaner;
+        this.securityContext = securityContext;
+        controllerManager.registerController("/admin/buildScanCleanup.html", this);
+        this.registerAction(this);
```

Comment:
Just for my understanding: why is registering the controller not enough, i.e. why is registerAction also needed?

Reply:
Looking at the Spring code, I now understand.